index.js

```diff
@@ -0,0 +1,72 @@
+'use strict';
+
+const Joi = require('joi');
+const emailer = require('./email');
+
+const SCHEMA_ADDRESS = Joi.string().email();
+const SCHEMA_ADDRESSES = Joi.array()
+    .items(SCHEMA_ADDRESS)
+    .min(0);
+const SCHEMA_STATUS = Joi.string();
+const SCHEMA_STATUSES = Joi.array()
+    .items(SCHEMA_STATUS)
+    .min(0);
+const SCHEMA_BUILD_SETTINGS = Joi.alternatives().try(
+    Joi.object().keys({ addresses: SCHEMA_ADDRESSES, statuses: SCHEMA_STATUSES }),
+    Joi.string().email()
+);
+const SCHEMA_BUILD_DATA = Joi.object()
+    .keys({
+        settings: SCHEMA_BUILD_SETTINGS,
+        status: SCHEMA_STATUS
+    });
+
+const SCHEMA_SMTP_CONFIG = Joi.object()
+    .keys({
+        host: Joi.string(),
+        port: Joi.number().integer(),
+        from: Joi.string().email()
+    });
+
+const DESCRIPTION_MAP = {
+    SUCCESS: 'Everything looks good!',
+    FAILURE: 'Did not work as expected.',
+    ABORTED: 'Aborted mid-flight',
+    RUNNING: 'Testing your code...',
+    QUEUED: 'Looking for a place to park...'
+};
+
+class EmailNotifier {
+    notify() {
```

Comment:
I understand that this isn't finished work. Do remember to add a function block sometime. It also helps us understand what you're trying to accomplish with this function -- there are some parts of this function that I have questions, but it's hard to distinguish whether that's intentional or an actual mistake.